server/src/test/java/org/elasticsearch/action/admin/indices/settings/put/UpdateSettingsRequestStreamableTests.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.settings.put;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.Settings.Builder;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.test.AbstractStreamableTestCase;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+import java.util.StringJoiner;
+
+public class UpdateSettingsRequestStreamableTests extends AbstractStreamableTestCase<UpdateSettingsRequest> {
+
+    @Override
+    protected UpdateSettingsRequest mutateInstance(UpdateSettingsRequest request) {
+        if (randomBoolean()) {
+            return new UpdateSettingsRequest(mutateSettings(request.settings()), request.indices());
+        }
+        return new UpdateSettingsRequest(request.settings(), mutateIndices(request.indices()));
+    }
+
+    @Override
+    protected UpdateSettingsRequest createTestInstance() {
+        return createTestItem();
+    }
+
+    @Override
+    protected UpdateSettingsRequest createBlankInstance() {
+        return new UpdateSettingsRequest();
+    }
+
+    @Override
+    protected void assertEqualInstances(UpdateSettingsRequest expectedInstance, UpdateSettingsRequest newInstance) {
+        newInstance.indices(expectedInstance.indices());
+        super.assertEqualInstances(expectedInstance, newInstance);
```

Comment:
shouldn't we rely on the ordinary equals/hashcode here? But also preserveExisting and indicesOptions should be taken into account?

Reply:
:facepalm: